src/BondingCurveRegistry.sol

```diff
@@ -38,52 +34,86 @@ contract BondingCurveRegistry is Initializable {
     // Mapping of curve addresses to curve IDs, for reverse lookup
     mapping(address => uint256) public curveIds;
 
+    // Mapping of the registered curve names, used to enforce uniqueness
+    mapping(string => bool) public registeredCurveNames;
+
+    // Address of the admin who may add curves to the registry
+    address public admin;
+
     /* =================================================== */
-    /*                    MODIFIERS                        */
+    /*                    EVENTS                           */
```

Comment:
3.14 - Missing Events in BondingCurveRegistry

We have added some core events here to the bonding curve registry.